source/backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -91,14 +93,34 @@ public PimsActivityInstance Add(PimsActivityInstance instance)
             return instance;
         }
 
-        public PimsActivityInstance Update(PimsActivityInstance instance)
+        public Tuple<PimsActivityInstance, PimsActivityInstanceNote> Update(PimsActivityInstance instance)
         {
             instance.ThrowIfNull(nameof(instance));
             var currentActivity = this.Context.PimsActivityInstances
                 .FirstOrDefault(x => x.ActivityInstanceId == instance.Id) ?? throw new KeyNotFoundException();
+
+            PimsActivityInstanceNote note = null;
+            if (currentActivity.ActivityInstanceStatusTypeCode!=instance.ActivityInstanceStatusTypeCode && this.User.HasPermission(Permissions.NoteAdd))
+            {
+                // add note
+                note = new PimsActivityInstanceNote()
+                {
+                    ActivityInstanceId = currentActivity.ActivityInstanceId,
+                    AppCreateTimestamp = System.DateTime.Now,
+                    AppCreateUserid = instance.AppCreateUserid,
+                    Note = new PimsNote()
+                    {
+                        IsSystemGenerated = true,
+                        NoteTxt = $"Activity status changed from { currentActivity.ActivityInstanceStatusTypeCode } to { instance.ActivityInstanceStatusTypeCode }",
+                        AppCreateTimestamp = System.DateTime.Now,
+                        AppCreateUserid = instance.AppCreateUserid,
+                    },
+                };
```

Comment:
Also want to double check what should be set for `AppCreateUserid`. You are setting it to the user that originally created the activity - which might not be the same as the current user making a status change, right?

The AC on JIRA says:

```
For the note: 
Record the audit fields same as for any other note (created by/on)
 - Created by - user who did the status change
 - Created on - the date on which the change happens
```

So I think this would record the wrong user. Say Bob created the activity but then Mary changes the status to "Archived". With the current logic the note would show as created by Bob when Mary was the one that did the actual change

Reply:
I think `AppCreateUserid` should be set to `this.User.Userid` to reflect who made the change...

Reply:
makes sense, using this.User.GetUsername() to capture current user trying to update.